fecgov/fec-cms
# Update wagtail to address pillow vulnerability

## Summary

- Resolves #3886 
- Resolves #3942 

Pinning the version of Pillow to 7.1.0 to address a Snyk vulnerability.

Snyk security update required an update to pillow but Wagtail 2.7.3 wouldn't allow that version.
Wagtail [increased the pillow limit to 8.0.0](https://github.com/wagtail/wagtail/commit/8939583e982fa0888bca1eccc5c8467f19d4e5d1) and that code made it to 2.7.4 so updating to 2.7.4 let us pin Pillow to 7.1.0.

Snyk alerts [574573](https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574573), [574574](https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574574), [574575](https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574575), [574576](https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574576), [574577](https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574577)


## Reviewers
- Someone from front-end,
- someone from content, (we'll make sure to tag you for review when it's ready to test outside of localhost)
- someone from design.
- Laura, I included you because you'd started this branch.

Feel free to remove yourselves.

## Impacted areas of the application

Technically the entire CMS but there should be no noticeable effect on presentation or admin

## Screenshots

No noticeable difference

## Related PRs

PR #3988 will address the Wagtail version upgrade for another vulnerability. We can disregard that one if this PR is merged instead.

## How to test

- Pull the branch like normal
- Install the upgraded Wagtail: inside pyenv, in the base cms directory, run `pip install -r requirements.txt`
- Should be no errors or warnings
- Check that you have Pillow 7.1.0: `pip show pillow`
- Run the CMS like normal: inside pyenv, in the fec directory, run `./manage.py runserver`
- Should be no errors or warnings
- Check that the localhost CMS works ([front-end](http://127.0.0.1) and [admin](http://127.0.0.1/admin/))
- Check a few different types of pages in both

____
